src/main/java/uk/ac/ebi/ega/egacryptor/cryptography/pgp/PGPCryptography.java

```diff
@@ -68,7 +68,7 @@ public InputStream decrypt(final InputStream fileToDecryptInputStream, final cha
     private OutputStream doEncrypt(final OutputStream fileOutputStream)
             throws IOException, PGPException {
         final OutputStream pgpEncryptedDataGeneratorOutputStream = PGPUtils.getEncryptedGenerator(pgpPublicKey).open(fileOutputStream, new byte[bufferSize]);
-        final PGPCompressedDataGenerator pgpCompressedDataGenerator = new PGPCompressedDataGenerator(PGPCompressedData.ZIP);
+        final PGPCompressedDataGenerator pgpCompressedDataGenerator = new PGPCompressedDataGenerator(CompressionAlgorithmTags.ZIP);
```

Comment:
PGPCompressedData implements CompressionAlgorithmTags and Sonar told me that I should use the variable from the parent class (CompressionAlgorithmTags), not from the subclass (PGPCompressedData).